content/200-concepts/100-components/01-prisma-schema/07-referential-actions.mdx

```diff
@@ -0,0 +1,327 @@
+---
+title: 'Referential actions'
+metaTitle: 'Referential actions'
+metaDescription: 'Referential actions let you define the update and delete behavior of related models on the database level'
+tocDepth: 2
+preview: true
+---
+
+<TopBlock>
+
+As of **2.26.0** Prisma now supports referential actions as part of the Prisma Schema Language.
+
+This is a [**preview feature**](../../../about/releases#preview) and can be accessed by:
+
+1. [Upgrading to version v2.26.0 of Prisma](../../../guides/upgrade-guides/upgrading-to-latest).
+2. Adding `referentialActions` to the `previewFeatures` field of the [`generator`](../../../reference/api-reference/prisma-schema-reference#generator) block in the `prisma.schema` file.
+
+```prisma file=schema.prisma
+generator client {
+    provider = "prisma-client-js"
+    previewFeatures: ['referentialActions']
+}
+```
+
+You now have 3 options:
+
+1. Make your schema reflect the current state of the database by running an [Introspection](../introspection). You can now see all relations in your data model using a different referential action compared to the new defaults.
+2. Define custom referential actions on your data model relation fields. You then update the database schema by running a [Migration](../prisma-migrate) or the [`db push`](../prisma-migrate/db-push) command.
+3. Use the new default referential actions that exist when no custom actions are added manually by running a [Migration](../prisma-migrate), or [`db push`](../prisma-migrate/db-push).
+
+To begin using referential actions you add them to the `@relation` attribute in your schema.
+
+```prisma
+@relation(fields: [authorId], references: [id], onDelete: Cascade)
```

Comment:
@relation on its own does not work, it required a field and type before. Do we usually show partial information like this or only in context of something that would work when you copy paste the whole line?

Reply:
Hmm, thats a good point. I thought it obvious what I was trying to show. But now that you mention it, I think its best to put it into full context. I was worried it would get lost in the schema but I can highlight the line